src/ctia/http/middleware/auth.clj

```diff
@@ -15,22 +16,23 @@
         (assoc-in [:headers "authorization"] auth-header))
     request))
 
-(defn testable-wrap-authentication
+(s/defn testable-wrap-authentication
   "wrap-autentication middleware."
-  [handler auth-service]
+  [handler
+   identity-for-token :- (s/=> s/Any s/Any)]
   (fn [request]
     (handler
      (if (:login request)
        request
        (let [auth-header (or (get-in request [:headers "authorization"])
                              (get-in request [:query-params "Authorization"]))
-             id (auth/identity-for-token auth-service auth-header)
+             id (identity-for-token auth-header)
              login (auth/login id)
              groups (auth/groups id)]
          (add-id-to-request request id login groups auth-header))))))
 
-(defn wrap-authentication [handler]
-  (testable-wrap-authentication handler @auth-service))
+(s/defn wrap-authentication [handler identity-for-token :- (s/=> s/Any s/Any)]
```

Comment:
```suggestion
(s/defn wrap-authentication [handler identity-for-token :- (s/pred ifn?)]
```

Reply:
I'm probably not going to delete the `s/=>`s if the only reason is the semantics.

There are two `identity-for-token` functions floating about, one protocol, one service dependency. IMO deleting this one loses documentation about which one we're using.

Using `s/=>` has been keeping me sane during this port since 80% of the changes are introducing HOF's!

Reply:
> I'm probably not going to delete the s/=>s if the only reason is the semantics.

I guess the reason is that they promise something that isn't delivered, and that can get out of sync. That can give us false confidence, leading us to confusion or mistakes as time passes.

Personally I'm OK with the => as a temporary tool for refactoring, or if there's a reasonable chance we'll get our own `=>` which actually performs checking.

Reply:
> I guess the reason is that they promise something that isn't delivered

(Warning: musing + opinions)

Fair enough. From my perspective, there are many ways of implementing `s/=>`, and to learn a contracts system is to learn which they choose. For example, with spec, you should know the differences between the semantics of `s/fspec` and `s/fdef` (or ignore them at your peril). Similarly, there's an analogous difference between `s/=>` and `s/defn` (technically the [docstrings](https://github.com/plumatic/schema/blob/ddb54c87dea6926c6d73542e517b53685cc82a37/src/cljx/schema/core.cljx#L1086-L1087) warn about these semantics, but obviously we all missed them).

In spec, the interesting semantics of `s/fspec` prevents me from using it in some cases, and overrides the usefulness of its documentation. OTOH `s/=>` has a conservative implementation that you could easily imagine in the future as being enhanced to "do the right thing" or rolling our own (as you say), so it's a clearer win to me in some cases.

My intention of pointing out the semantics of `s/=>` was to help the team better understand schema, which we certainly do now. I didn't anticipate a purge of `s/=>` definitions, but that's a side-effect of the better understanding, so I think my mission is accomplished (I thought it might stimulate interest in adding semantics to `s/=>` if anything).

A somewhat relevant note: I've found it useful to combine `s/=>` with `s/fn` in the body of functions to be explicit about both the interface of a function and its checking semantics. eg.,

```clojure
(s/defn might-return-a-fn :- (s/maybe (s/=> s/Int s/Int))
  []
  (s/fn :- s/Int [i :- s/Int] (+ 1 i)))
```

Reply:
Honestly I have a hard time seeing your POV, after a few attempts 😃

Would you maintain the same opinion if `=>` checked `some?` instead of `ifn?`? Or if it did no checking at all?

From the [docstring](https://github.com/plumatic/schema/blob/ddb54c87dea6926c6d73542e517b53685cc82a37/src/cljx/schema/core.cljx#L1072), which uses the word `currently` I perceive a hint that there wasn't a strong philosophy behind its current implementation; a stronger implementation surely was left in a wishlist that never materialised.

Reply:
In any case, I value the documentation aspect of `=>`.